pkg/registry/events/event/strategy.go

```diff
@@ -56,8 +61,13 @@ func (eventStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Obje
 
 // Validate validates a new Event.
 func (eventStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
-	event := obj.(*api.Event)
-	return validation.ValidateEvent(event)
+	groupVersion := requestGroupVersion(ctx)
+	if groupVersion == eventsv1api.SchemeGroupVersion {
+		event := obj.(*eventsv1.Event)
```

Comment:
We're calling validation on internal representation of the object. Which for eventsv1 is still coreapi.Event.

So this will simply panic.
[The reason why it is not panicking is that noone is using it - the events library is still using eventsv1beta1.]

Basically you should also change to:
```
event := obj.(*coreapi.Event) 
```
here to (or rather do the casting before the if once).

And that also mean migrating the ValidateEvent to actually use coreapi.Event as an input

Reply:
Done.